### ISSUE TITLE
Crowdsale contract (LPers and KOLs)

### ISSUE THREAD

**Issue by 0xViva** [wzrdx/pumpichu#16]

**Comment by strainz420** [comment]
i wonder if we can just ditch the openzepplin crowdsale stuff and use our own.
we should not use solidity 4 haha
a lot of security stuff is in v8, e.g. not requiring SafeMath stuff.

**Comment by 0xViva** [comment]
I agree, getting OZ 2.5 with solidity version 5 didnt feel very good. This is also simply gonna be a presale/distribution contract. So we could probably figure something out by just looking at the old crowdsale.sol logic.

**Comment by b10c77** [comment]
You can probably compile without changes by changing the solidity version to a newer one

**Comment by 0xViva** [comment]
> You can probably compile without changes by changing the solidity version to a newer one

I'll try this tomorrow.
